enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,583 @@
+installations. The baremetal platform solves the "floating ingress" issue by
+using virtual IP addresses/keepalived.
+
+- Deal with users who want their ingress traffic to not go through the single
+control-plane node. Such users can easily achieve that goal by modifying the
+`IngressController` to target just the worker, non-control-plane node(s)
+(perhaps by labeling all of them with a unique label that is not applied to the
+control-plane node that is also a worker and then have the `IngressController`
+target that label).
+
+- Deal with non-cloud, non-"none" platforms such as baremetal, vSphere, etc.
+
+## Proposal
+
+- Create a new `DefaultPlacement` API field.
+
+- Make sure worker node ignition files are generated even in bootstrap-in-place
+single control-plane node "none"-platform installation.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want to install a single control-plane
+node cluster with additional worker nodes, so that my cluster will be able to
+handle larger computation demands.
+
+- As an OpenShift cluster administrator, I want to add worker nodes to my
+existing single control-plane node cluster, so that it'll be able to meet
+growing computation demands.
+
+### API Extensions
+
+Introduce a new status field in the Ingress config CR
+(`config.openshift.io/v1/ingresses`) called `DefaultPlacement`.
+
+In addition, continue to allow the `.spec.replicas` and `.spec.nodePlacement`
+fields in `operator.openshift.io/v1/ingresscontrollers` CRs to be omitted, but
+change the "controller-defaulting" behavior for these fields to be  based on
+this new status field. This means that when the Cluster Ingress Operator
+creates router `Deployments` out of `IngressController` resources it will
+adjust how it behaves in the absence of these parameters. The exact details of
+this behavior are described below.
+
+The goal of this new `DefaultPlacement` field is to give the installer a way to
+affect the general placement of the `IngressController`. This is regardless of
+whether an `IngressController` is created during installation with the
+`.spec.replicas` and `.spec.nodePlacement` omitted or whether no
+`IngressController` was created during an installation at all. Today, when no
+`IngressController` is created during installation, the Cluster Ingress Operator
+creates a default `IngressController`, in this case we can go one of two ways -
+
+- The Cluster Ingress Operator leaves those fields empty in the newly created
+`IngressController`
+
+- The cluster Ingress Operator populates these fields in the newly created
+`IngressController`
+
+Choosing one way or the other is arbitrary and shouldn't matter due to the
+immutable nature of this new field - the eventual `Deployment` that gets
+created would be identical in any case. I personally prefer the second option
+since it makes the fields explicit in the `IngressController` itself and not
+just in `Deployments` that get created.
```

Comment:
I think there are differences, if you leave the fields empty, the controller can change the values at runtime later, if you have the controller populate the values (this should be via admission webhook), then the values are now set and can not be changed later unless the user does so. I was under the impression we wanted to go with the first option of leaving them empty for now to maintain existing behaviour?

Reply:
Note that what I'm discussing here is specifically what the controller would do to the IngressController it **creates itself**, as can be seen in [this](https://github.com/openshift/cluster-ingress-operator/blob/0f36470250c3089769867ebd72e25c413a29cda2/pkg/operator/operator.go#L218-L254) code - it's a `ensureDefaultIngressController` function that creates an IngressController if one was not created during installation - whether it bakes the replicas/placement into that particular IngressController or not is of no consequence, due to the immutability of the new defaultPlacement parameter